core/src/main/java/com/graphhopper/routing/util/NameSimilarityEdgeFilter.java

```diff
@@ -19,85 +19,183 @@
 
 import com.graphhopper.util.EdgeIteratorState;
 import com.graphhopper.apache.commons.lang3.StringUtils;
+import info.debatty.java.stringsimilarity.JaroWinkler;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.util.regex.Pattern;
 
 /**
- * Abstract Class that defines the basis for NameSimilarity matching using an EdgeFilter.
+ * This Class defines the basis for NameSimilarity matching using an EdgeFilter.
```

Comment:
Would you add some more explanation here? E.g. *to snap coordinates not necessarily to its closest edge but instead pick an edge with a similar name we need to define similary and this happens here*

minor typo: Class->class